File: ExpenseSplitter.Frontend/ExpenseSplitter.Web/src/app/models/expense/create-expense-model.ts
import { ExpenseTypeEnum } from 'src/app/data/expense-type';
import { ExpensePartModel } from './expense-part-model';

export class CreateExpenseModel {
    name: string;
    type: ExpenseTypeEnum;
    paidAt: Date;
    payerId: number;

    parts: ExpensePartModel[];
}

File: ExpenseSplitter.Backend/ExpenseSplitter.Api/Models/Expenses/CreateExpenseModel.cs

using System;
using System.Collections.Generic;
using ExpenseSplitter.Api.Data;

namespace ExpenseSplitter.Api.Models.Expenses
{
    public class CreateExpenseModel
    {
        public string Name { get; set; }
        public ExpenseType Type { get; set; }
        public DateTime PaidAt { get; set; }
        public int PayerId { get; set; }

        public List<ExpensePartModel> Parts { get; set; }
    }
}
